lib/matplotlib/backend_bases.py

```diff
@@ -3413,6 +3413,12 @@ def add_toolitem(self, name, group, position, image, description, toggle):
               state between consecutive clicks).
             * `False` : The button is a normal button (returns to unpressed
               state after release).
+
+        Note
+        ----
+        This method is not intended to be called by end-users (who should
+        instead `.ToolContainerBase.add_tool`); it is provided to be
+        implemented by third-party backends.
```

Comment:
I would phrase the whole docstring differently:

```
A hook to add a toolitem to the container.

This hook must be implemented in each backend and contains the backend-specific
code to add a tool; e.g. add an element to the toolbar.

.. warning::
    This is part of the backend implementation and should not be called by end-users.
    They should instead call `.ToolContainerBase.add_tool`.
```

also for the other cases

Reply:
Sure, done.